diff --git a/CMakeLists.txt b/CMakeLists.txt
index 314d5adaa8..19e59b1728 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -208,8 +208,6 @@ endif()
 # Compiler / Linker flags
 if("${PLATFORM}" STREQUAL "DARWIN")
   jerry_add_link_flags(-lSystem)
-  set(CMAKE_C_ARCHIVE_CREATE "<CMAKE_AR> Sqc <TARGET> <LINK_FLAGS> <OBJECTS>")
-  set(CMAKE_C_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
   set(CMAKE_SHARED_LINKER_FLAGS  "-undefined dynamic_lookup")
 elseif((NOT CYGWIN AND NOT MINGW AND NOT MSYS) AND (USING_GCC OR USING_CLANG))
   jerry_add_link_flags(-Wl,-z,noexecstack)
